### PR TITLE
Update/4.0.0

### PR DESCRIPTION
**Pull request by krugazul**
### Requirements

Please review this branch of LSX Theme. We would like to know if there are any issues with the folder structure or naming conventions, or any coding practices you disagree with.

#### Removed

- All PHP templates and template parts.
- All functionality relating to actions hooks and filters ( what our plugins use to output )
- All vendor and 3rd party and Vendor libraries

The only functionality that was left over was our "welcome" screen, which you see after activating the theme. Everything else was removed.   As well as some optimisation and schema classes I will be working on.

#### Added

- Added in block and site templates for the post single and archives. (block template parts as well). Everything needed to display.
- Started on a basic [custom field block](https://github.com/lightspeeddevelopment/lsx/tree/update/4.0.0/blocks) which saves the info added into it, into a custom field.
- Added in a basic theme.json and a few variables - [theme.json](https://github.com/lightspeeddevelopment/lsx/blob/update/4.0.0/theme.json) - Justin has been working in a [child theme](https://github.com/lightspeeddevelopment/lsdev-frostwp-child/blob/main/theme.json) with some more advanced usage.

### CODE REVIEW

File: blocks/build/post-meta/index.css [hunk 1]
```diff
@@ -0,0 +1 @@
+.wp-block-lsx-post-meta-basic input[type=text]{background:inherit;border:0;color:inherit;font-family:Gilbert,sans-serif;font-size:64px}
```
Comment (fklein-lu) [inline blocks/build/post-meta/index.css:1]: It's preferable not to have any styles for basic elements in a block. Instead blocks should rely on the styles provided by the theme.

File: blocks/src/post-meta/save.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * be combined into the final markup, which is then serialized by the block
+ * editor into `post_content`.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#save
+ *
+ * @param {Object} props            Properties passed to the function.
+ * @param {Object} props.attributes Available block attributes.
+ * @return {WPElement} Element to render.
+ */
+export default function save( { attributes } ) {
```
Comment (fklein-lu) [inline blocks/src/post-meta/save.js:20]: This block is static, and does not do what the `block.json` indicates it does.

File: parts/footer.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<!-- wp:group {"style":{"spacing":{"padding":{"top":"var(--wp--custom--spacing--large, 8rem)"}}},"layout":{"inherit":true}} -->
+<div class="wp-block-group" style="padding-top:var(--wp--custom--spacing--large, 8rem)">
+<!-- wp:pattern {"slug":"twentytwentytwo/footer-default"} /-->
```
Comment (fklein-lu) [inline parts/footer.html:3]: These patterns need to be updated, seems like a copy-paste error.

File: theme.json [hunk 4]
```diff
@@ -0,0 +1,140 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/theme.json",
```
Comment (fklein-lu) [inline theme.json:2]: It's best to align the `theme.json` schema with the current version of WordPress.

**Review by fklein-lu — CHANGES_REQUESTED** [review]

File: blocks/build/post-meta/index.css (hunk 1 above)
Comment (krugazul) [reply blocks/build/post-meta/index.css]: Ive removed the styles, they are not needed for this block.  I will make sure to rely on the styles provided by the theme.
 - [commit](https://github.com/lightspeeddevelopment/lsx/commit/b59d0e5b6a8729fb03810f6466d20a5c9eab2b01)

**Comment by krugazul** [comment]
We will not be merging this just yet.  We are going too work in the branch, and branch off of that with smaller pull requests.

**Pull request closed without merging**